### ISSUE TITLE
Inconsistent behaviour with data/dates storage

### ISSUE THREAD

**Issue by lahus2** [Statoil/sunbeam#75]
Feel free to change title based on your understanding and close this if I am wrong. @joakim-hove asked me to look at changes in https://github.com/Statoil/sunbeam/pull/72

I tested this briefly and compared results between komodo versions testing and bleeding. The changes seem to have fixed the date structure (KEYWORDS attached to date before them not after them) but possibly introduced another bug.

script used
````
#!/usr/bin/env python 

from sunbeam.tools import TimeVector
import inspect
import datetime

tv=TimeVector(datetime.date(1999,12,31))
tv.load('TEMPLATE.SCH', date=datetime.datetime(1999,12,31))

for i in range(len(tv)):
  print '\n' + str(i) + ' ---------------------------------'
  print tv[i]
print '\nDATES ---------------------------------'
print tv.dates
````

schedule file for testing
```
RPTSCHED
   RESTART=2 /

WRFTPLT
'*' YES /
/

DATES
1 'JAN' 2000 /
/

RPTSCHED
   RESTART=0 /

DATES
1 'FEB' 2000 /
/


DATES
1 'MAR' 2000 /
/
````

print of tv (should match input schedule in this case and does with testing version of komodo)
```
DATES
  31 'DEC' 1999/
/

RPTSCHED
   'RESTART=2' /

WRFTPLT
   '*' 'YES' /
/

DATES
  1 'JAN' 2000/
/

RPTSCHED
   'RESTART=0' /

DATES
  1 'MAR' 2000/
/
````

There is an extra DATES (31/12/1999) at start of file that will potentially cause issues with Eclipse and a missing date (1/2/2000). We add DATES with no keywords to force results to be calculated at this point. Preserving this information is very important.  Let me know if you need more information.

**Comment by joakim-hove** [comment]
OK - this:
```python
tv.load('TEMPLATE.SCH')
```
did not work - got an error message which does make some sense - IMHO. However - this should work:
```python
tv = TimeVector(datetime.date(1999,12,31, base_file = "TEMPLATE.SCH")
```

**Comment by lahus2** [comment]
I wanted to clarify one point about date at start. How can I load a file without a date appearing at the top. For example with a real case we have a data file that includes a start date then further dates in schedule file.

DATA file paraphrased
-START in data file
- INCLUDE XXX.SCH /

XXX.SCH paraphrased
- WELSPECS/COMPDAT/.WCONPROD/...
- DATES ...

The first date in schedule file cannot be equal to start data in data file. Is it possible with sunbeam to load a full schedule file and work with it (without printing the DATES with start date) at top? When I tested with komodo testing version before the changes to dates that keywords attached to the file was replicated perfectly.

**Comment by joakim-hove** [comment]
OK - just to be clear: the current implementation is a best effort, if it is not good enough/what you want we change it!

> How can I load a file without a date appearing at the top. 

When you load a file/string you have two options:

1. The file already starts with `DATES` keyword.
2. You supply a value for the optional `date=` keyword when calling `load`.

> XXX.SCH paraphrased

The way it is meant to be used - might be a bad idea - is that if you have a rough 90% correct schedule file which you to manipulate; that should be supplied as an argument to the `TimeVector`constructor. I we have the situation:

1. I have a schedule file - roughly speaking a *base case* in `BASE.SCH`.
2. I want to add keywords to the fiile at specific dates, then I can either load a file part with a leading `DATES` keyword, or alternatively pass a `date=` argument to the load function.


**file_with_DATES.txt**
```txt
DATES 
    10 'MAR' 2020 /

WELSPECS
   'NEW-WELL' ...
/
```
**keyword_only.txt**
```txt
COMPDAT
    -- ADD MORE COMPLETIONS  ...
/
```

This can then be loaded with:
```python
tv = TimeVector(datetime.date(2000,1,1), base_file = "BASE.SCH")
tv.load("file_with_DATES.txt")
tv.load("keyword_only.txt" , date=datetime(2025,1,1))
```

The important point here is that we start with skeleton from `BASE.SCH`. Does this make sens?

**Comment by joakim-hove** [comment]
Hopefully fully fixed in #76
